### PR TITLE
Add note about indexing config classes

### PR DESCRIPTION
**Pull request by alexandre-daubois**
Fix https://github.com/symfony/symfony-docs/issues/19156

This is a recurring question as far as I know in the repo issues so let's add a note about it? :slightly_smiling_face:

### CODE REVIEW

**Review by 94noni — APPROVED** [review]
nice addition indeed

File: configuration.rst [hunk 1]
```diff
@@ -1304,6 +1304,11 @@ namespace ``Symfony\Config``::
     Nested configs (e.g. ``\Symfony\Config\Framework\CacheConfig``) are regular
     PHP objects which aren't autowired when using them as an argument type.
 
+.. note::
+
+    Be sure to not exclude the ``kernel.build_dir`` directory from your IDE
```
Comment (ghost) [inline configuration.rst:1309]: Very nice addition. This is exactly what I've stumbled on today. 
However the problem is that by default `kernel.build_dir` equals to cache directory. I don't think devs want to have this dir indexed. On the other hand even if build dir is different then I wouldn't like to have all of it indexed as well.

What I did in my IDE (PHPStorm) is to mark a subdir `var/cache/dev/Symfony/Config` as not excluded while the `var` dir remains marked as excluded.

Do you think we can be specific about this in this note?

Comment (alexandre-daubois) [reply configuration.rst:1309]: Il afraid it's a bit specific 🤔 I think we should someone from @symfony/team-symfony-docs

Comment (javiereguiluz) [reply configuration.rst:1309]: I was divided about this ... but at the end I added a explicit mention to this directory (https://github.com/symfony/symfony-docs/commit/a3fe10800a2b92596c19f341d5cfa1bd4eba09ab). It's true that it's very specific ... but this note is specifically talking about that, so I think it's fine. Thanks!

**Comment by javiereguiluz** [comment]
Thanks Alex! I love when contributors add changes that solve recurring questions from the community.

**Pull request merged**